app/assets/scripts/components/map.js

```diff
@@ -185,8 +186,9 @@ const Map = React.createClass({
     }
 
     this.overlay = L.geoJson(regions, { style }).bindPopup(function ({ feature }) {
-      const id = feature.properties.admin_id;
-      const name = isDistrict ? districtNameMap[id] : get(byEgy(id), 'name');
+      const id = isDistrict ? feature.properties.id : feature.properties.admin_id;
+      const name = isDistrict ? get(byIdDist(id)) : get(byEgyGove(id), 'name');
```

Comment:
Seems like `get(byIdDist(id))` is either redundant or should be `get(byIdDist(id), 'name')`